include/swift/Syntax/Serialization/SyntaxSerialization.h

```diff
@@ -199,6 +200,253 @@ struct NullableTraits<RC<syntax::RawSyntax>> {
   }
 };
 } // end namespace json
+
+namespace byteTree {
+
+template <>
+struct WrapperTypeTraits<tok> {
+  static uint8_t numericValue(const tok &Value) {
+    // Explicitly spell out all toks to keep the serialized value stable
+    // even if its members get reordered or members get removed
```

Comment:
Instead of explicitly spell out them all in the code, how about adding "code" property in `gyb_syntax_support` nodes? like [`include/swift/AST/Attr.def`](https://github.com/apple/swift/blob/master/include/swift/AST/Attr.def)

Reply:
Something like

```python
% for idx, kind in enumerate(SYNTAX_KINDS):
  case TokenKind::${kind}: return ${idx}
```

so you don't have to manually specify them?

Reply:
@harlanhaskins We want stable code for each kinds those aren't changed even if we add/delete kinds. We can use `enumerate` *only if* we ensure to increment version number every time we add/delete node kind.

Reply:
@rintaro Adding a `code` property to `gym_syntax_support` will work quite well for token kinds because all of them are defined in one file `Token.py`. I'll adopt it for this. This should definitely be an improvement.

I don't, however, think that it's a good idea to add the code for the `SyntaxKind`s in `gyb_syntax_support` since the different node types are defined in many different files and finding the next free code would be fairly difficult since you need to look at all declared nodes. What do you think?

Reply:
How about adding centralized dictionary in `gyb_syntax_support`? like:
```python
SYNTAX_KIND_CODES = {
  'Token': 0,
  'Unknown': 1
  'UnknownDecl': 2,
  'TypealiasDecl': 3,
  ...
}
```
At least, we can share it between C++ and SwiftSyntax. Also, we can add verification check in `gym_syntax_support/__init__.py` or somewhere so we ensure every `SyntaxKind` has a code assigned.